src/Microsoft.OData.Core/Json/JsonReader.cs

```diff
@@ -1489,19 +1516,85 @@ private void TryPopPropertyScope()
         /// <summary>
         /// Skips all whitespace characters in the input.
         /// </summary>
-        /// <returns>true if a non-whitespace character was found in which case the tokenStartIndex is pointing at that character.
-        /// false if there are no non-whitespace characters left in the input.</returns>
+        /// <returns>true if a non-whitespace character was found; false if end of input was reached.</returns>
         private bool SkipWhitespaces()
         {
             do
             {
-                for (; this.tokenStartIndex < this.storedCharacterCount; this.tokenStartIndex++)
+                int remaining = this.storedCharacterCount - this.tokenStartIndex;
+                int vectorSize = Vector<ushort>.Count;

```

Comment:
@WanjohiSammy My suggestion is to switch to `ReadOnlySpan<char>.IndexOfAnyExcept` with a cached `SearchValues<char>` of the 4 JSON whitespace chars (' ', '\t', '\r', '\n' - **only**). This is already vectorized by the runtime and keeps the code tiny. I think it should be at least as fast and much easier to maintain across x64/ARM64.

If custom vectorization logic should remain, add a small-span fast-path (e.g. < 16 or < 32 chars) to avoid paying setup cost on short inputs where SIMD can be slower than scalar

Again, if the custom handwritten logic should remain, factor the span-scan into one helper to avoid duplication and divergence between sync and async flavours - if possible.